csi/moac/volume.ts

```diff
@@ -490,14 +454,41 @@ export class Volume {
   async create() {
     log.debug(`Creating the volume "${this}"`);
 
+    this.attach();
+
     // Ensure there is sufficient number of replicas for the volume.
     const newReplicaCount = this.replicaCount - Object.keys(this.replicas).length;
     if (newReplicaCount > 0) {
       // create more replicas if higher replication factor is desired
       await this._createReplicas(newReplicaCount);
     }
-    this._setState(VolumeState.Healthy);
+    const [nexusNode, replicaSet] = await this._ensureReplicaShareProtocols();
+    // If the nexus poped up while we were creating replicas pick it up now.
+    // Though it's an unsual situation so we log a warning if it happens.
+    if (!this.nexus) {
+      const nexus = this.registry.getNexus(this.uuid);
+      if (nexus) {
+        log.warn(
+          `The nexus "${nexus}" appeared while creating replicas - using it`
+        );
+        this.newNexus(nexus);
```

Comment:
Is it possible that the nexus that we are now using has a different number of replicas and therefore we don't have the same configuration as if we were to create a new one?

Reply:
I think that might be possible. We have already created replicas and ensured the share protocols on them and then suddenly a strange nexus pops up. What is the chance that this will work if we don't know anything about the nexus? FSA would fix any inconsistencies however it is non-deterministic as we don't know when and IO could fail in the meantime if volume is used. Maybe this part of the code is a bit over-thought and we should just go straight to the nexus creation without checking if it exists. AFAIK the create will fail in that case and we simply return an error to the CSI method caller. If you think so, I will remove this part of code to make it simpler.

Reply:
I think I would prefer what you suggested - to attempt to create the nexus and return a failure if it fails. At least that way we know what we're getting. Plus, from what I understand, the chances of hitting this particular case is small, so we shouldn't be seeing nexus create failures often.

Reply:
Agree. I will make this change 👍